mlrun/projects/project.py

```diff
@@ -2817,13 +2818,31 @@ def _set_function(
 
         self.spec.set_function(name, function_object, func)
 
+    # TODO: Remove this in 1.10.0
+    @deprecated.deprecated(
+        version="1.8.0",
+        reason="'remove_function' is deprecated and will be removed in 1.10.0. "
+        "Please use `delete_function` instead.",
+        category=FutureWarning,
+    )
     def remove_function(self, name):
         """remove the specified function from the project
 
         :param name:    name of the function (under the project)
         """
         self.spec.remove_function(name)
 
+    def delete_function(self, name, from_cache_only=False):
```

Comment:
Since this is replacing the existing `remove_function` we should maintain the same default logic, i.e. only delete from client-side project by default. So, I'd change `from_cache_only` to be `True` by default.
As a matter of fact, I think it's better since this is the new functionality to call this parameter `delete_from_db` and reverse its meaning (and of course have it `False` by default).